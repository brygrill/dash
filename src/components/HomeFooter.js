// @flow
import React, { Component } from 'react';
import RaisedButton from 'material-ui/RaisedButton';

import swim from '../images/icons/swim_circle_white.svg';
import bike from '../images/icons/bike_circle_white.svg';
import run from '../images/icons/run_circle_white.svg';

const iconStyle = {
  height: '5rem',
  width: '5rem',
};
export default class HomeFooterComponent extends Component {
  props: {
    btnLabel: string,
    onBtnClick: Function,
  };

  render() {
    return (
      <div
        className="mdl-cell mdl-cell--12-col mdl-cell--middle"
        style={{ color: '#fff', textAlign: 'center' }}
      >
        <div className="mdl-grid">
          <div className="mdl-cell mdl-cell--12-col">
            <img src={swim} alt="swim icon" style={iconStyle} />
            <img src={bike} alt="bike icon" style={iconStyle} />
            <img src={run} alt="run icon" style={iconStyle} />
          </div>
          <div className="mdl-cell mdl-cell--12-col">
            <RaisedButton
              label={this.props.btnLabel}
              secondary
              onTouchTap={this.props.onBtnClick}
            />
          </div>
        </div>
      </div>
    );
  }
}
