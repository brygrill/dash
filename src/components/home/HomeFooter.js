// @flow
import React, { Component } from 'react';
import RaisedButton from 'material-ui/RaisedButton';

import swim from '../../images/icons/swim_circle_white.svg';
import bike from '../../images/icons/bike_circle_white.svg';
import run from '../../images/icons/run_circle_white.svg';

const iconStyle = {
  height: '5rem',
  width: '5rem',
};
export default class HomeFooterComponent extends Component {
  static defaultProps = {
    disabled: false,
  };

  props: {
    btnLabel: string,
    disabled?: boolean,
    onBtnClick: Function,
  };

  render() {
    return (
      <div
        className="mdl-cell mdl-cell--12-col mdl-cell--middle"
        style={{ color: '#fff', textAlign: 'center' }}
      >
        <div className="mdl-grid">
          <div className="mdl-cell mdl-cell--12-col">
            <img src={swim} alt="swim icon" style={iconStyle} />
            <img src={bike} alt="bike icon" style={iconStyle} />
            <img src={run} alt="run icon" style={iconStyle} />
          </div>
          <div className="mdl-cell mdl-cell--12-col">
            <RaisedButton
              label={this.props.btnLabel}
              secondary
              disabled={this.props.disabled}
              onTouchTap={this.props.onBtnClick}
            />
          </div>
        </div>
      </div>
    );
  }
}
