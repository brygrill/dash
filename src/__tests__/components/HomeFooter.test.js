import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import HomeFooter from '../../components/home/HomeFooter';

describe('<HomeFooter />', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MuiThemeProvider><HomeFooter /></MuiThemeProvider>, div);
  });
  it('Matches snapshot with child', () => {
    const component = renderer.create(
      <MuiThemeProvider><HomeFooter /></MuiThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
