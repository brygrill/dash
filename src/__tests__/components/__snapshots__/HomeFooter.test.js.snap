// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeFooter /> Matches snapshot with child 1`] = `
<div
  className="mdl-cell mdl-cell--12-col mdl-cell--middle"
  style={
    Object {
      "color": "#fff",
      "textAlign": "center",
    }
  }
>
  <div
    className="mdl-grid"
  >
    <div
      className="mdl-cell mdl-cell--12-col"
    >
      <img
        alt="swim icon"
        src="swim_circle_white.svg"
        style={
          Object {
            "height": "5rem",
            "width": "5rem",
          }
        }
      />
      <img
        alt="bike icon"
        src="bike_circle_white.svg"
        style={
          Object {
            "height": "5rem",
            "width": "5rem",
          }
        }
      />
      <img
        alt="run icon"
        src="run_circle_white.svg"
        style={
          Object {
            "height": "5rem",
            "width": "5rem",
          }
        }
      />
    </div>
    <div
      className="mdl-cell mdl-cell--12-col"
    >
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                   0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#ff4081",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
              "zIndex": 1,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
